truffle/src/main/java/org/jruby/truffle/nodes/ext/BigDecimalNodes.java

```diff
@@ -677,218 +805,297 @@ public Object multSpecial(RubyBasicObject a, RubyBasicObject b) {
+                            return createBigDecimal(frame, Type.POSITIVE_INFINITY);
                         case 0:
-                            return a;
+                            return createBigDecimal(frame, Type.POSITIVE_INFINITY);
                         case -1:
-                            return createRubyBigDecimal(Type.NEGATIVE_INFINITY);
+                            return createBigDecimal(frame, Type.NEGATIVE_INFINITY);
                     }
                 case NEGATIVE_INFINITY:
                     switch (getBigDecimalValue(b).signum()) {
                         case 1:
-                            return a;
+                            return createBigDecimal(frame, Type.NEGATIVE_INFINITY);
                         case 0:
-                            return a;
+                            return createBigDecimal(frame, Type.NEGATIVE_INFINITY);
                         case -1:
-                            return createRubyBigDecimal(Type.POSITIVE_INFINITY);
+                            return createBigDecimal(frame, Type.POSITIVE_INFINITY);
                     }
             }
             CompilerAsserts.neverPartOfCompilation();
             throw new UnsupportedOperationException();
         }
 
-        @Specialization(guards = {
-                "!isNormal(a)",
-                "isRubyBigDecimal(b)",
-                "!isNormal(b)"})
-        public Object divSpecialSpecia(RubyBasicObject a, RubyBasicObject b) {
+        protected Object divSpecialSpecial(VirtualFrame frame, RubyBasicObject a, RubyBasicObject b, int precision) {
             final Type aType = getBigDecimalType(a);
             final Type bType = getBigDecimalType(b);
 
             if (aType == Type.NAN || bType == Type.NAN ||
                     (aType == Type.NEGATIVE_ZERO && bType == Type.NEGATIVE_ZERO))
-                return createRubyBigDecimal(Type.NAN);
+                return createBigDecimal(frame, Type.NAN);
 
             if (aType == Type.NEGATIVE_ZERO)
-                if (bType == Type.POSITIVE_INFINITY) return a;
-                else return createRubyBigDecimal(Type.POSITIVE_INFINITY);
+                if (bType == Type.POSITIVE_INFINITY) return createBigDecimal(frame, Type.NEGATIVE_ZERO);
+                else return createBigDecimal(frame, Type.POSITIVE_INFINITY);
 
             if (bType == Type.NEGATIVE_ZERO)
-                if (aType == Type.POSITIVE_INFINITY) return createRubyBigDecimal(Type.NEGATIVE_INFINITY);
-                else return createRubyBigDecimal(Type.POSITIVE_INFINITY);
+                if (aType == Type.POSITIVE_INFINITY) return createBigDecimal(frame, Type.NEGATIVE_INFINITY);
+                else return createBigDecimal(frame, Type.POSITIVE_INFINITY);
 
             // a and b are only +-Infinity
-            return createRubyBigDecimal(Type.NAN);
+            return createBigDecimal(frame, Type.NAN);
+        }
+    }
+
+    @CoreMethod(names = {"/", "quo"}, required = 1)
```

Comment:
Spaces between the brace and the quotation mark.  We still have code that doesn't have the space.   It's not critical, but we're trying to normalize moving forward.